godot-ffi/src/global.rs

```diff
@@ -57,72 +62,30 @@ impl<T> Global<T> {
     /// If the initialization function panics. Once that happens, the global is considered poisoned and all future calls to `lock()` will panic.
     /// This can currently not be recovered from.
     pub fn lock(&self) -> GlobalGuard<'_, T> {
-        let mutex_guard = self
-            .value
-            .lock()
-            .expect("Global<T> poisoned; a thread has panicked while holding a lock to it");
-
-        let guard = Self::ensure_init(mutex_guard, true)
-            .unwrap_or_else(|| panic!("previous Global<T> initialization failed due to panic"));
-
-        guard
+        let guard = self.value.lock().unwrap_or_else(PoisonError::into_inner);
+        guard.get_or_init(self.init_fn);
+        // SAFETY: above line guarantees init
+        unsafe { GlobalGuard::new_unchecked(guard) }
```

Comment:
Are we sure the `get_or_init()` isn't more expensive than the previous `Self::ensure_init()`?

While it may not matter in average cases, it would be a pity to pessimize performance just to make our own life a bit easier.

Reply:
Nitpick, please also leave an empty line before `//` comment blocks (here `// SAFETY`).

Reply:
the `get_or_init` here boils down to
```rust
if self.get().is_some() { return };
let value = f();
let slot = unsafe { &mut *self.inner.get() };
slot.insert(value);
```
which seems like an improvement from the previous `ensure_init` fn.

and ive benchmarked it to be about the same speed.